File: frontend/src/types/project.ts
// ────────────────────────────────────────────────────────────────────────────
// Project Management – TypeScript Type Definitions
// Each Project represents one client website.
// ────────────────────────────────────────────────────────────────────────────

export type ProjectStatus = "active" | "onboarding" | "maintenance" | "archived";

export interface Project {
  id: string;
  organization_id: string;
  creator_id?: string | null;
  name: string;
  slug: string;
  website_url?: string | null;
  country?: string | null;
  timezone?: string | null;
  industry?: string | null;
  currency: string;
  status: ProjectStatus;
  logo_url?: string | null;
  description?: string | null;
  created_at: string;
  updated_at: string;
}

export interface ProjectAuditLog {
  id: string;
  action: string;
  resource_type: string;
  resource_id?: string | null;
  details?: Record<string, unknown> | null;
  created_at: string;
}

export interface ProjectDetail extends Project {
  audit_logs: ProjectAuditLog[];
}

export interface ProjectListResponse {
  items: Project[];
  total: number;
  page: number;
  limit: number;
  total_pages: number;
}

export interface ProjectCreateRequest {
  name: string;
  website_url?: string;
  country?: string;
  timezone?: string;
  industry?: string;
  currency?: string;
  status?: ProjectStatus;
  logo_url?: string;
  description?: string;
}

export interface ProjectUpdateRequest {
  name?: string;
  website_url?: string;
  country?: string;
  timezone?: string;
  industry?: string;
  currency?: string;
  status?: ProjectStatus;
  logo_url?: string;
  description?: string;
}

export interface ProjectListParams {
  search?: string;
  status?: string;
  industry?: string;
  country?: string;
  sort_by?: string;
  sort_order?: "asc" | "desc";
  page?: number;
  limit?: number;
}

// ── Static Options ────────────────────────────────────────────────────────────

export const PROJECT_STATUSES: { value: ProjectStatus; label: string; color: string }[] = [
  { value: "active", label: "Active", color: "emerald" },
  { value: "onboarding", label: "Onboarding", color: "blue" },
  { value: "maintenance", label: "Maintenance", color: "amber" },
  { value: "archived", label: "Archived", color: "slate" },
];

export const PROJECT_INDUSTRIES = [
  "SaaS",
  "E-commerce",
  "Healthcare",
  "Finance",
  "Education",
  "Real Estate",
  "Logistics",
  "Media",
  "Gaming",
  "Retail",
  "Manufacturing",
  "Other",
];

export const PROJECT_CURRENCIES = [
  { code: "USD", label: "US Dollar ($)" },
  { code: "EUR", label: "Euro (€)" },
  { code: "GBP", label: "British Pound (£)" },
  { code: "PKR", label: "Pakistani Rupee (₨)" },
  { code: "AED", label: "UAE Dirham (د.إ)" },
  { code: "INR", label: "Indian Rupee (₹)" },
  { code: "CAD", label: "Canadian Dollar (C$)" },
  { code: "AUD", label: "Australian Dollar (A$)" },
];

export const TIMEZONES = [
  "UTC",
  "America/New_York",
  "America/Chicago",
  "America/Denver",
  "America/Los_Angeles",
  "America/Toronto",
  "Europe/London",
  "Europe/Paris",
  "Europe/Berlin",
  "Asia/Karachi",
  "Asia/Dubai",
  "Asia/Kolkata",
  "Asia/Singapore",
  "Asia/Tokyo",
  "Australia/Sydney",
  "Pacific/Auckland",
];

File: backend/app/schemas/project.py
import uuid
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field, HttpUrl


class ProjectCreateRequest(BaseModel):
    """Client website project creation request DTO."""

    name: str = Field(..., min_length=2, max_length=255)
    website_url: Optional[str] = Field(None, max_length=255)
    country: Optional[str] = Field(None, max_length=100)
    timezone: Optional[str] = Field(None, max_length=100)
    industry: Optional[str] = Field(None, max_length=100)
    currency: str = Field("USD", min_length=3, max_length=10)
    status: str = Field("active", max_length=50)
    logo_url: Optional[str] = Field(None, max_length=500)
    description: Optional[str] = None


class ProjectUpdateRequest(BaseModel):
    """Client website project update request DTO."""

    name: Optional[str] = Field(None, min_length=2, max_length=255)
    website_url: Optional[str] = Field(None, max_length=255)
    country: Optional[str] = Field(None, max_length=100)
    timezone: Optional[str] = Field(None, max_length=100)
    industry: Optional[str] = Field(None, max_length=100)
    currency: Optional[str] = Field(None, min_length=3, max_length=10)
    status: Optional[str] = Field(None, max_length=50)
    logo_url: Optional[str] = Field(None, max_length=500)
    description: Optional[str] = None


class AuditLogResponse(BaseModel):
    """Audit log entry summary for project timeline."""

    id: uuid.UUID
    action: str
    resource_type: str
    resource_id: Optional[str] = None
    details: Optional[dict] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class ProjectResponse(BaseModel):
    """Client website project detail response DTO."""

    id: uuid.UUID
    organization_id: uuid.UUID
    creator_id: Optional[uuid.UUID] = None
    name: str
    slug: str
    website_url: Optional[str] = None
    country: Optional[str] = None
    timezone: Optional[str] = None
    industry: Optional[str] = None
    currency: str = "USD"
    status: str = "active"
    logo_url: Optional[str] = None
    description: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectDetailResponse(ProjectResponse):
    """Project details including audit logs timeline."""

    audit_logs: List[AuditLogResponse] = []


class ProjectListResponse(BaseModel):
    """Paginated projects response wrapper."""

    items: List[ProjectResponse]
    total: int
    page: int
    limit: int
    total_pages: int
